hack/generator/pkg/astmodel/file_writer.go

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package astmodel
+
+import (
+	"bufio"
+	"io"
+	"os"
+
+	"github.com/dave/dst"
+	"github.com/dave/dst/decorator"
+)
+
+// FileAst represents a file of Go code
+type FileAst interface { // TODO: Is there a better name? Something about CodeGeneratedFile?
+	// AsAst transforms the file into a dst.File (you can think of this as the AST representation of the file)
+	AsAst() (*dst.File, error)
+}
+
+// This is just a grouping mechanism currently -- we could
+// possibly have some configuration here but we don't right now
+type FileWriter struct{}
```

Comment:
If you made this `GoSourceFileWriter` with a single member like this:
```
type GoSourceFileWriter struct {
    file GoSourceFile
}
```
and a factory method like this:
``` go
func NewGoSourceFileWriter(file GoSourceFile) { ... }
```
then you could avoid always having to pass the file into the methods.

Reply:
I was originally thinking that the writer would be usable for multiples files and might hold some configuration state around things like if it should `go fmt` the files before it wrote them or whatnot. In the abscence of any actual need for that though, I'll move to the model you suggested for now and we can always refactor back to this if it turns out we have some cross-cutting configuration we want to store.

Reply:
Sounds good.